### PR TITLE
Fix asset generation for .csproj .NET Core projects

### LINKED ISSUE #767: Update launch.json generation logic for MSBuild-based .NET Core
The current launch.json generation logic needs to be updated to handle MSBuild-based .NET Core. Currently, it only runs on project.json. This requires a few things to happen:

- [x] OmniSharp needs to be updated to understand multiple target frameworks in .csproj ([example](https://github.com/dotnet/sdk/blob/8f09baf9b3c600d7df3495c6eb68814959b4bafb/TestAssets/TestProjects/CrossTargeting/DesktopAndNetStandard/DesktopAndNetStandard.csproj)) files using the latest MSBuild bits.
- [ ] The OmniSharp `/projects` end point needs to be updated to return information for multiple target frameworks (like it does for project.json projects).
- [x] The OmniSharp `/projects` end point needs to be updated to include output file path information.
- [x] The C# extension needs to be updated to a newer version of OmniSharp with the changes above.
- [ ] The launch.json/tasks.json generation logic needs tweaked to work for .csproj projects.

### PR DESCRIPTION
**Pull request by DustinCampbell**
Fixes #767

### CODE REVIEW

File: src/assets.ts [hunk 1]
```diff
@@ -52,22 +50,284 @@ interface AttachConfiguration extends DebugConfiguration {
...
+            version: '0.1.0',
+            command: 'dotnet',
+            isShellCommand: true,
+            args: [],
+            tasks: [this.createBuildTaskDescription()]
+        };
+    }
+}
+
+function findNetCoreAppTargetFramework(project: protocol.MSBuildProject) {
```
Comment (gregg-miskelly) [inline src/assets.ts]: nit: I always like having a return type when it isn't obvious.

File: src/assets.ts [hunk 2]
```diff
@@ -52,22 +50,284 @@ interface AttachConfiguration extends DebugConfiguration {
...
+
+function containsDotNetCoreProjects(workspaceInfo: protocol.WorkspaceInformationResponse) {
+    if (workspaceInfo.DotNet && findExecutableProjectJsonProjects(workspaceInfo.DotNet.Projects, 'Debug').length > 0) {
+        return true;
+    }
+
+    if (workspaceInfo.MsBuild && findExecutableMSBuildProjects(workspaceInfo.MsBuild.Projects).length > 0) {
+        return true;
+    }
+}
```
Comment (gregg-miskelly) [inline src/assets.ts]: Do you need a `return false;` here?

Comment (DustinCampbell) [inline src/assets.ts]: What? It's JavaScript. Of course not!

And, yes, I clearly missed that. Thanks!

File: src/assets.ts (hunk 1 above)
Comment (DustinCampbell) [inline src/assets.ts]: Thanks!

**Pull request merged**